blosc/blosc2.c

```diff
@@ -2302,6 +2299,17 @@ int blosc_compress_context(blosc2_context* context) {
   else {
     // Check whether we have a run for the whole chunk
     int start_csizes = context->header_overhead + 4 * context->nblocks;
+    int dont_split = (context->header_flags & 0x10) >> 4;
+    int nstreams = context->nblocks;
+    if (!dont_split) {
+      // When splitting, the number of streams is computed differently
+      if (context->leftover) {
+        nstreams = (context->nblocks - 1) * context->typesize + 1;
+      }
+      else {
+        nstreams *= context->typesize;
+      }
+    }
```

Comment:
I above code should be put in its original position (see comment above).

Reply:
solved at f2fdffa71814dd1cbda6d3388e551c7e61214f64